### PR TITLE
[VEN-1929]: force liquidation functionality in diamond implementation of comptroller

### PR DESCRIPTION
**Pull request by Debugger022**
## Description

Resolves 
- VEN-1929: force liquidation implementation in diamond Comptroller architecture 

## Checklist

- [ ] I have updated the documentation to account for the changes in the code.
- [ ] If I added new functionality, I added tests covering it.
- [ ] If I fixed a bug, I added a test preventing this bug from silently reappearing again.
- [ ] My contribution follows [Venus contribution guidelines](docs/CONTRIBUTING.md).

### CODE REVIEW

File: tests/hardhat/Comptroller/Diamond/comptrollerTest.ts [hunk 1]
```diff
@@ -534,5 +592,144 @@ describe("Comptroller", () => {
...
+            accounts[0].address,
+            root.address,
+            convertToUnit("1", 18),
+          );
+          expect(errCode).to.equal(3);
+        });
+
+        it("succeeds if borrower has nonzero shortfall", async () => {
+          vToken.borrowBalanceStored.returns(convertToUnit("100", 18));
+          comptrollerLens.getHypotheticalAccountLiquidity.returns([0, 0, 1]);
```
Comment (kkirka) [inline tests/hardhat/Comptroller/Diamond/comptrollerTest.ts:722]: Shouldn't we remove lens here and fake oracle price instead? AFAIU we don't use lens in Diamond Comptroller anymore

Comment (Debugger022) [reply tests/hardhat/Comptroller/Diamond/comptrollerTest.ts:722]: we are using lens in the FacetBase, https://github.com/VenusProtocol/venus-protocol/blob/feat/diamond-proxy/contracts/Comptroller/Diamond/facets/FacetBase.sol#L141

File: contracts/Comptroller/Diamond/facets/SetterFacet.sol [hunk 2]
```diff
@@ -494,4 +497,13 @@ contract SetterFacet is ISetterFacet, FacetBase {
         minReleaseAmount = minReleaseAmount_;
         emit NewVAIVaultInfo(vault_, releaseStartBlock_, minReleaseAmount_);
     }
+
+    function _setForcedLiquidation(address vTokenBorrowed, bool enable) external {
+        ensureAllowed("setForcedLiquidation(address,bool)");
```
[the remark this answers is not in the document]
Comment (Debugger022) [reply contracts/Comptroller/Diamond/facets/SetterFacet.sol]: fixed

File: contracts/Comptroller/ComptrollerStorage.sol
Comment (chechu) [inline contracts/Comptroller/ComptrollerStorage.sol:1]: `venusSpeeds` and `venusRate` were converted into `internal` functions [here](https://github.com/VenusProtocol/venus-protocol/pull/332/files)

Comment (Debugger022) [reply contracts/Comptroller/ComptrollerStorage.sol:1]: Fixed, 35b5517efac7618203782cfb6ff4f4dc0a8e0345

**Review by chechu — APPROVED** [review]

**Pull request merged**